starfish/spots/_detector/local_search_blob_detector.py

```diff
@@ -0,0 +1,423 @@
+            spot in anchor_round.
+        channels, rounds : Sequence[int]
+            Channels and rounds present in the ImageStack from which spots were detected.
+        search_radius : int
+            The maximum (euclidean) distance in pixels for a spot to be considered matching in
+            a round subsequent to the anchor round.
+        anchor_round : int
+            The imaging round to seed the local search from.
+
+        """
+
+        anchor_df = round_dataframes[anchor_round]
+
+        # create empty IntensityTable filled with np.nan
+        data = np.full((dist.shape[0], len(channels), len(rounds)), fill_value=np.nan)
+        dims = (Features.AXIS, Axes.CH.value, Axes.ROUND.value)
+        coords = {
+            Features.SPOT_RADIUS: (Features.AXIS, anchor_df[Features.SPOT_RADIUS]),
+            Axes.ZPLANE.value: (Features.AXIS, anchor_df[Axes.ZPLANE]),
+            Axes.Y.value: (Features.AXIS, anchor_df[Axes.Y]),
+            Axes.X.value: (Features.AXIS, anchor_df[Axes.X]),
+            Axes.ROUND.value: (Axes.ROUND.value, rounds),
+            Axes.CH.value: (Axes.CH.value, channels)
+        }
+        intensity_table = IntensityTable(data=data, dims=dims, coords=coords)
+
+        # fill IntensityTable
+        for r in rounds:
+
+            # get intensity data and indices
+            spot_indices = ind[r]
+            intensity_data = round_dataframes[r].loc[spot_indices, 'intensity']
+            channel_index = round_dataframes[r].loc[spot_indices, Axes.CH]
+            round_index = np.full(ind.shape[0], fill_value=r, dtype=int)
+            feature_index = np.arange(ind.shape[0], dtype=int)
+
+            # mask spots that are outside the search radius
+            mask = np.asarray(dist[r] < search_radius)  # indices need not match
+            feature_index = feature_index[mask]
+            channel_index = channel_index[mask]
+            round_index = round_index[mask]
+            intensity_data = intensity_data[mask]
+
+            # need numpy indexing to set values in vectorized manner
+            intensity_table.values[feature_index, channel_index, round_index] = intensity_data
+
+        return intensity_table
+
+    def run(
+        self, data: ImageStack, verbose: bool=False, n_processes: Optional[int]=None
+    ) -> IntensityTable:
+        """Find 1-hot coded spots in data.
+
+        Parameters
+        ----------
+        data : ImageStack
+            Image data containing coded spots.
+        verbose : bool
+            If True, report on progress of spot finding.
+        n_processes : Optional[int]
+            Number of processes to devote to spot finding. If None, will use the number of available
+            cpus (Default None).
+
+        Returns
+        -------
+        IntensityTable
+            Contains detected coded spots.
+
+        """
+        per_tile_spot_results = self._find_spots(data, verbose=verbose, n_processes=n_processes)
```

Comment:
`_find_spots` returns `Dict[Tuple[int, int], np.ndarray]`
`_merge_spots_by_round` accepts a `Dict[Tuple[int, int], pd.DataFrame]`

one of these is wrong. :)